### PR TITLE
Add constant_pad_nd to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/helpers.h [hunk 1]
```diff
@@ -48,6 +49,17 @@ class XlaHelpers {
     return xla::ConstantLiteral(builder, ScalarLiteral(scalar_value, type));
   }
 
+  static xla::XlaOp ScalarValue(const at::Scalar& scalar_value,
+                                xla::PrimitiveType type,
+                                xla::XlaBuilder* builder) {
+    if (scalar_value.isFloatingPoint()) {
+      return ScalarValue(scalar_value.toDouble(), type, builder);
+    }
+    XLA_CHECK(scalar_value.isIntegral()) << "Scalar type not supported";
```
Comment (dlibenzi) [inline torch_xla/csrc/helpers.h]: Stream the scalar as well is at::Scalar supports it.

File: torch_xla/csrc/ops/constant_pad_nd.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class ConstantPadNd : public Node {
+ public:
+  ConstantPadNd(const Value& input,
+                tensorflow::gtl::ArraySlice<const xla::int64> pad,
+                const at::Scalar& value);
+
+  std::string ToString() const override;
+
+  XlaOpVector Lower(LoweringContext* loctx) const override;
+
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/constant_pad_nd.h]: pad() missing.

File: torch_xla/csrc/helpers.h (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/helpers.h]: Only for integral and double, not complex (for example).

File: torch_xla/csrc/ops/constant_pad_nd.h (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/ops/constant_pad_nd.h]: Done.

**Pull request merged**